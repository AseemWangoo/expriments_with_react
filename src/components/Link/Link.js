import { useState } from "react";
import PropTypes from "prop-types";

const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal",
};

const Link = ({ page, children }) => {
  const [status, setStatus] = useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <a
      className={status}
      href={page || "#"}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
};

Link.defaultProps = {
  page: "https://github.com/aseemwangoo",
};

Link.propTypes = {
  page: PropTypes.string,
};

export default Link;
