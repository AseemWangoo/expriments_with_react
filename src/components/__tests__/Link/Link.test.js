import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Link from "../../Link/Link";

afterEach(() => {
  cleanup();
});

test("renders correctly", () => {
  const component = renderer.create(
    <Link page="https://github.com/aseemwangoo">Made with ❤️ - Aseem</Link>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
