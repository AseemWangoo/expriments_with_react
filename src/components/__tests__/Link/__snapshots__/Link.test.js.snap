// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="https://github.com/aseemwangoo"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Made with ❤️ - Aseem
</a>
`;
